FilecoinFoundationWeb/filecoin-foundation
# [UXIT-1100] - Include Nested Checkbox for Ecosystem Subcategories [skip percy]

> [!NOTE]
> The high number of file changes is mostly content files. Only 14 file changes are code-related.

## 📝 Description

This PR adds subcategory filters to the ecosystem page on mobile and desktop.

The first two commits update the `FormCheckbox` and `SlideOver` components to adapt them for this feature:
- 10f0ce9945852fe1be85772cd19f7674f1f136cd (1 file changed)
- 6c0a508255865105aa5e132f25bd53cee542990b (1 file changed)

The third commit is where the implementation of the feature is:
- 44c2e7df7f8730605f479288e29c24c5d48e720e (12 files changed)

The fourth and last commit cleans the ecosystem project Markdown entries. There were many mismatches between the subcategories and their parent category, so I manually updated them. We might want to enforce that at the Zod level going forward, if feasible.
- 16e7d170d883a52c6647e0a79969ee303a9810bb (35 files changed)

## 🛠️ Key Changes
- Created `getCategoriesFromDirectory` and `getSubcategoriesFromDirectory` to fetch the list of categories and subcategories from the CMS config.
- Used `Zod` to validate the data coming from the CMS config, as we are increasingly going in this direction
- Added `SubcategoryFilters` with `nuqs` to sync the selected checkboxes with the URL.
- Created `useEcosystemCategory`, which replaces `useCategory` on the ecosystem explorer page

> [!IMPORTANT]
> The vertical spacing between checkboxes is bigger than on Figma on smaller devices to account for touch targets. @filipagr approved the design with me via DM.  

## 🧪 How to Test

1. Go to `/ecosystem-explorer`
2. Apply some filters and make sure the Results total and selected projects match the selection

## 📸 Screenshots

![CleanShot 2024-11-19 at 09 31 47](https://github.com/user-attachments/assets/df159cc1-062a-4ec4-920d-c8fae94f7898)

## 🔖 Resources

- https://nuqs.47ng.com/docs/parsers/built-in#arrays
